src/app/code/community/Omise/Gateway/Model/PaymentMethod.php

```diff
@@ -135,4 +135,47 @@ public function assignData($data)
 
         return $result;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see app/code/core/Mage/Payment/Model/Method/Abstract.php
+     * @see Mage_Payment_Model_Method_Abstract::canUseForCurrency()
+     */
+    public function canUseForCurrency($currencyCode)
+    {
+        switch ($currencyCode) {
+            case 'THB':
+            case 'JPY':
+            case 'IDR':
+            case 'SGD':
+                return true;
+                break;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param  string          $currency
+     * @param  integer | float $amount
+     *
+     * @return integer
+     */
+    public function formatAmount($currency, $amount)
+    {
+        switch (strtoupper($currency)) {
+            case 'THB':
+            case 'IDR':
+            case 'SGD':
+                // Convert to a small unit
+                $amount = $amount * 100;
+                break;
+
+            case 'JPY':
```

Comment:
@guzzilar If the currency does not require formatting, I think this condition can be removed.

For example, if another currency has been opened to receive payment and it does not require formatting. To comply with this pattern, it need to add another empty condition.

For an example with Philippine peso,

```php
case 'JPY':
    break;

case 'PHP':
    break;
```

And maybe more currencies.